stock_reserve_sale/model/sale.py

```diff
@@ -0,0 +1,160 @@
+# Copyright 2013 Camptocamp SA - Guewen Baconnier
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import api, fields, models
+from odoo.exceptions import UserError
+from odoo.osv import expression
+from odoo.tools.translate import _
+
+_LINE_KEYS = ["product_id", "product_uom_qty"]
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    @api.depends(
+        "state", "order_line.reservation_ids", "order_line.is_stock_reservable"
+    )
+    def _compute_stock_reservation(self):
+        for sale in self:
+            has_stock_reservation = False
+            is_stock_reservable = False
+            for line in sale.order_line:
+                if line.reservation_ids:
+                    has_stock_reservation = True
+                if line.is_stock_reservable:
+                    is_stock_reservable = True
+            if sale.state not in ("draft", "sent"):
+                is_stock_reservable = False
+            sale.is_stock_reservable = is_stock_reservable
+            sale.has_stock_reservation = has_stock_reservation
+
+    has_stock_reservation = fields.Boolean(
+        compute="_compute_stock_reservation",
+        readonly=True,
+        multi="stock_reservation",
+        store=True,
+        string="Has Stock Reservations",
+    )
+    is_stock_reservable = fields.Boolean(
+        compute="_compute_stock_reservation",
+        readonly=True,
+        multi="stock_reservation",
+        store=True,
+        string="Can Have Stock Reservations",
+    )
+    reserves_count = fields.Integer(compute="_compute_reserves_count")
+    all_lines_reserved = fields.Boolean(
+        compute="_compute_reserves_count", store=True, default=False
+    )
+
+    def release_all_stock_reservation(self):
+        line_ids = [line.id for order in self for line in order.order_line]
+        lines = self.order_line.browse(line_ids)
```

Comment:
Hello, thanks for your works.
I think we can optimize this function.
[EDIT 11/09] Adds @gdgellatly suggestion instead of self.mapped("order_line")
```suggestion
        lines = self.order_line
```

Reply:
not just `self.order_line`?

Reply:
If self is a recordset, i don't remember if odoo gives all order_line for all record without a mapped.

Reply:
@FrancoMaxime In v16, you can call the relation field on multi recordset

Reply:
@rousseldenis Thanks